### ISSUE TITLE
Cosmic upgrade / unity - 7.5 - no more icons on the dekstop and no right click menu

### ISSUE THREAD

**Issue by jetboo** [shiznix/unity-gentoo#211]
After upgrading to Cosmic my icons on the desktop disappeared, also there "right click" menu is not showing up anymore. Please tell me where I could look at to find the issue.
Cheers

**Comment by c4pp4** [comment]
Try to logout and login back, or restart, or `dconf reset -f /org/compiz/`

**Comment by jetboo** [comment]
> `dconf` reset -f `/org/compiz/` + logout/login did not help.
Is there a log file where I can see what is wrong ?

**Comment by c4pp4** [comment]
`journalctl -b` (last boot messages)

**Comment by c4pp4** [comment]
What about if you try to create test user account and try to login into this to confirm it's not user settings issue.

**Comment by jetboo** [comment]
> What about if you try to create test user account and try to login into this to confirm it's not user settings issue.

It is the same problem, no Dekstop icons, no right-click menu.

The logout button doesn't work as well. When chosen to logout, I end up without any menu, nor launch-bar icons, just the plain wallpaper and it remains stuck. The reboot/shutdown works fine. 

here is the journalctl -b command http://dpaste.com/3ZTNN87 
I can't understand each one of these lines but these two seems weird to me

`Mar 18 19:04:23 canbox zeitgeist-datah[2708]: zeitgeist-datahub.vala:207: Error during inserting events: GDBus.Error:org.gnome.zeitgeist.EngineError.InvalidArgument: Incomplete event: interpretation, manifestation and actor are required `
`
Mar 18 19:04:27 canbox compiz[2629]: console.error: CustomizableUI:`

Cheers

**Comment by c4pp4** [comment]
I haven't found anything wrong. I have the same error message about zeitgeist.

It looks like org.gnome.desktop.background show-desktop-icons is false, you can try `gsettings set org.gnome.desktop.background show-desktop-icons true`, but I don't think it's the case.

Reproduce logout stuck, wait at least 2 minutes and paste journal. If you need second to last boot messages, use `journalctl -b -1` (and so on -2, -3, ...)

**Comment by jetboo** [comment]
> gsettings set org.gnome.desktop.background show-desktop-icons true

Does not work neither. Journal: http://dpaste.com/15T8QV5

**Comment by shiznix** [comment]
I have had it happen on occasion too, about every 1 in 10 desktop launches.

In my case it is nautilus-desktop that is segfaulting.
Which makes sense as it's Nautilus that is drawing the desktop icons and background menu.

I'll try and get a useful backtrace from coredumpctl...

**Comment by jetboo** [comment]
> I have had it happen on occasion too, about every 1 in 10 desktop launches.
> 
> In my case it is nautilus-desktop that is segfaulting.
> Which makes sense as it's Nautilus that is drawing the desktop icons and background menu.
> 
> I'll try and get a useful backtrace from coredumpctl...

Wait, I dont have any nautilus-dekstop running, a ps ux |grep nautilus-dekstop returns nothing 

```
ps ux |grep nautilus-dekstop                                                                                      
can      11010  0.0  0.0 133216   984 pts/9    S+   11:46   0:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn nautilus-dekstop
```

and it doesn't seemed installed (no auto-completion in my terminal). What package provides it ? nautilus ? 

and emerge -av nautilus:
```
emerge -av nautilus

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   #] gnome-base/nautilus-3.30.5::gentoo  USE="gnome introspection previewer (seccomp) -gtk-doc (-packagekit) (-selinux) -sendto -test" 0 KiB
```

**Comment by shiznix** [comment]
Means it must be uninstalled (emerge -C gsettings-qt) before it will install.

**Comment by shiznix** [comment]
> Wait, I dont have any nautilus-dekstop running, a ps ux |grep nautilus-dekstop returns nothing

> ps ux|grep nautilus-desktop                                                                                 
> can      12846  0.0  0.0 133216   916 pts/1    S+   12:00   0:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn nautilus-desktop
> ```
> and it doesn't seemed installed (no auto-completion in my terminal). What package provides it ? nautilus ?
> 
> and emerge -av nautilus:
> 
> ```
> emerge -av nautilus
> 
> These are the packages that would be merged, in order:
> 
> Calculating dependencies... done!
> [ebuild   R   #] gnome-base/nautilus-3.30.5::gentoo  USE="gnome introspection previewer (seccomp) -gtk-doc (-packagekit) (-selinux) -sendto -test" 0 KiB
> ```

It won't be running if it's segfaulted.
It's provided by gnome-base/nautilus and should be at /usr/bin/nautilus-desktop

**Comment by shiznix** [comment]
But I notice your installed gnome-base/nautilus is not the overlay version - should be 3.26.4_p_p0_p0701

**Comment by jetboo** [comment]
> Are you using the overlay's Cosmic profile?

Yes:
```
eselect profile list
[46]  unity-gentoo:linux/amd64/17.0/cosmic (stable) *
```

**Comment by c4pp4** [comment]
> [ebuild R #] gnome-base/nautilus-3.30.5::gentoo

according to "#" sign you unmasked gentoo main tree nautilus package, maybe more packages... look at /etc/portage/package.unmask

**Comment by jetboo** [comment]
I cleaned my package.unmask, un-merged a bunch of gnome-3.30 stuff. Switched to gcc-8.2, "depclean"-ed all the unity-gentoo packages (this was maybe a bit extreme) and remerged everything again.

Now I have the right nautilus 
```
emerge -av nautilus

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   ~] gnome-base/nautilus-3.26.4_p_p0_p0701::unity-gentoo  USE="exif gnome introspection previewer (-packagekit) (-selinux) -sendto -test -tracker -xmp" 0 KiB
```
and I got my icons and the desktop back. The right-click menu works as well.
